src/Core/Domain/Configuration/ShopConfigurationInterface.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Configuration;
+
+use PrestaShop\PrestaShop\Core\ConfigurationInterface;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+
+interface ShopConfigurationInterface extends ConfigurationInterface
+{
+    /**
+     * @param string $key
+     * @param mixed|null $default
+     * @param ShopConstraint|null $shopConstraint
+     *
+     * @return mixed
+     */
+    public function get($key, $default = null, ShopConstraint $shopConstraint = null);
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     * @param ShopConstraint|null $shopConstraint
+     *
+     * @return ShopConfigurationInterface
+     */
+    public function set($key, $value, ShopConstraint $shopConstraint = null);
+
+    /**
+     * @param string $key
+     * @param ShopConstraint|null $shopConstraint
+     *
+     * @return bool
+     */
+    public function has($key, ShopConstraint $shopConstraint = null);
```

Comment:
```suggestion
    public function get(string $key, $default = null, ShopConstraint $shopConstraint = null);

    /**
     * @param string $key
     * @param mixed $value
     * @param ShopConstraint|null $shopConstraint
     *
     * @return ShopConfigurationInterface
     */
    public function set(string $key, $value, ShopConstraint $shopConstraint = null): ShopConfigurationInterface;

    /**
     * @param string $key
     * @param ShopConstraint|null $shopConstraint
     *
     * @return bool
     */
    public function has(string $key, ShopConstraint $shopConstraint = null): bool;
```

Reply:
Nope, for the second time ^^ it's not possible that's the one I wanted to add (especially `string` type for `$key`)
But it's not possible because `ConfigurationInterface` (the interface we are extending) is type less, so adding strict type results in an error 

```php
Fatal error: Declaration of PrestaShop\PrestaShop\Core\Domain\Configuration\ShopConfigurationInterface::get(string $key, $default = NULL, ?PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint $shopConstraint = NULL) must be compatible with PrestaShop\PrestaShop\Core\ConfigurationInterface::get($key) in /Users/jo/www/prestashop-177/src/Core/Domain/Configuration/ShopConfigurationInterface.php on line 34
```

Reply:
The best we can do is having correct PHPDoc and wait for the next major to add appropriate strict type
Besides for the return type, if we wanted to allow extending the interface (but maybe we won't need any more) we'd better return `self` instead of a strict type (same in the parent interface)